### PR TITLE
Feature/23 add search page

### LINKED ISSUE #23: Add search page
TBD:
1. Show "To search page" button to header autocomplete if in header search not empty value
2. If no search query show "Search for locations to get quick access to weather information for the locations you care about."
3. on enter search query and pressing 'enter' button search result showed as a list
4. On click list item behavior same as in autocomplete 
5. Page has loading and no data state ("We couldn't find any locations matching’.
Check that you spelled the location name correctly.
Try searching for another location.")

### PR DESCRIPTION
**Pull request by Avareco**

### CODE REVIEW

File: src/components/Header.svelte [hunk 1]
```diff
@@ -1,49 +1,23 @@
...
-  import { clickOutside } from '../directives/clickOutside';
-  import { debounce } from 'lodash';
   import { t, locale } from 'svelte-i18n';
-  import {
-    addToHistory,
-    getLocationsHistoryAutocomplete,
-    removeFromHistory,
-  } from '../utils/searchHistory';
-  import { NOTIFICATION_TYPE, notificationsStore } from '../stores/notification';
+  import ForecastAutocomplete from './ForecastSearch.svelte';
```
Comment (oleksandrstarov) [inline src/components/Header.svelte]: Any reason to rename component on import?

File: src/routes/Search.svelte [hunk 2]
```diff
@@ -1 +1,73 @@
-<div>Search</div>
+<script>
```
Comment (oleksandrstarov) [inline src/routes/Search.svelte]: Probably it is a good idea to name route components  as pages - SearchPage in this case

File: src/components/Header.svelte (hunk 1 above)
Comment (Avareco) [reply src/components/Header.svelte]: Missed that. will fix

File: src/routes/Search.svelte (hunk 2 above)
Comment (Avareco) [reply src/routes/Search.svelte]: Good point, lets rename

File: src/components/ForecastSearch.svelte [hunk 3]
```diff
@@ -0,0 +1,231 @@
+<script>
+  import { t } from 'svelte-i18n';
+  import { Button, Input } from 'flowbite-svelte';
+  import {
+    addToHistory,
+    getLocationsHistoryAutocomplete,
+    removeFromHistory,
+  } from '../utils/searchHistory';
+  import { location, push, querystring } from 'svelte-spa-router';
```
Comment (Aleshchenk0) [inline src/components/ForecastSearch.svelte]: It seems to me we don't use location here.

File: src/i18n/ua.json [hunk 4]
```diff
@@ -18,14 +19,23 @@
     "metersPerSecond": "м/с",
     "forecastMainBanner.noDataReceived": "Дані не отримані"
   },
+  "search": {
+  "searchResults": "Результати пошуку",
+  "searchResultsCount": "Показано {count} результатів для '{query}':",
```
Comment (Aleshchenk0) [inline src/i18n/ua.json]: Please, remove redundant quotes around `{query}`

**Comment by Aleshchenk0** [comment]
As I figured out you decided to refuse getting suggested places vis `Enter` click, yeah?

**Comment by Avareco** [comment]
> As I figured out you decided to refuse getting suggested places vis `Enter` click, yeah?

No it's there(str 120, ForecastSearch) and navigation to first autocomplete still works. If you mean this functionality  on search page I suppose it will be strange as search results it is not autocomplete

**Comment by Aleshchenk0** [comment]
> > As I figured out you decided to refuse getting suggested places vis `Enter` click, yeah?
> 
> No it's there(str 120, ForecastSearch) and navigation to first autocomplete still works

I see. I ment after pressing `Enter` could shown list of suggestions.

**Review by Aleshchenk0 — APPROVED** [review]

**Review by oleksandrstarov — APPROVED** [review]
Looks good

**Pull request merged**